signer/src/stacks_api.rs

```diff
@@ -40,154 +36,324 @@ impl StacksClient {
     /// StacksSettings.
     pub fn new(settings: StacksSettings) -> Self {
         Self {
-            api_endpoint: settings.api.endpoint,
             node_endpoint: settings.node.endpoint,
             client: reqwest::Client::new(),
         }
     }
 
-    /// Get Stacks block IDs given the bitcoin block hash. Uses the Stacks API
-    /// via the GET /extended/v2/burn-blocks/:height_or_hash endpoint.
-    ///
-    /// See https://docs.hiro.so/api/get-burn-block
-    async fn get_block_ids(&self, hash: &bitcoin::BlockHash) -> Result<Vec<StacksBlockId>, Error> {
-        // The Stacks API expects the hash to be hex encoded with the
-        // leading 0x in the string, which is not produced by the Display
-        // implementation of bitcoin::hashes::sha256d::Hash (but is for
-        // the Debug implementation).
-        let hash: &bitcoin::hashes::sha256d::Hash = hash.as_raw_hash();
-        let path = format!("/extended/v2/burn-blocks/0x{}", hash);
-        let url = self.api_endpoint.join(&path).map_err(Error::PathParse)?;
-
-        tracing::debug!(%hash, "Fetching block IDs confirmed by bitcoin block from stacks API");
+    /// Fetch the raw stacks nakamoto block from a Stacks node given the
+    /// Stacks block ID.
+    async fn get_block(&self, block_id: StacksBlockId) -> Result<NakamotoBlock, Error> {
+        let path = format!("/v3/blocks/{}", block_id.to_hex());
+        let base = self.node_endpoint.clone();
+        let url = base
+            .join(&path)
+            .map_err(|err| Error::PathJoin(err, base, Cow::Owned(path)))?;
+
+        tracing::debug!(%block_id, "Making request to the stacks node for the raw nakamoto block");
```

Comment:
Since the block ID is a parameter to the function, we don't need to explicitly add it to the log call if we instrument the function. I think that makes the code a bit cleaner.
```suggestion
        tracing::debug!("Making request to the stacks node for the raw nakamoto block");
```